lib/grant.js

```diff
@@ -449,7 +452,24 @@ function saveRefreshToken (done) {
 }
 
 /**
- * Create an access token and save it with the model
+ * Pass the scope string to the model for saving
+ *
+ * @param  {Function} done
+ * @this   OAuth
+ */
+function saveScope (done) {
+  var scope = this.scope || this.req.body.scope;
+
+  var self = this;
+  this.model.saveScope(scope, self.accessToken, function (err, acceptedScope) {
+    if (err) return done(error('server_error', false, err));
```

Comment:
@nunofgs: Yeah, I encountered the same issue when I first wrote that part, and I couldn't really find a clean workaround. Nowhere else in the library are we handling "cases" of returned errors, so it would feel awkward to introduce something of the sort here.

The only thing I can think of is to have a dedicated model method, such as `validateScope`. I'm writing some more notes on this topic in the PR #149 discussion in a moment, including the saveAuthCode matter.

The scope middleware is utilized for access validation _once a token has already been created_, it does not influence how scopes are first processed and stored.